### ISSUE TITLE
SPARKs unable to change loadout while wounded

### ISSUE THREAD

**Issue by Iridar** [X2CommunityCore/X2WOTCCommunityHighlander#1121]
This was a bug report under one of my mods, so 70% chance it's bogus, but if it's not, it's a pretty big deal, since being able to take SPARKs on mission even if they're damage is their shtick. 

Opening an issue so I don't forget to investigate.

**Comment by Iridar** [comment]
Further information received from MrCloista:

https://www.youtube.com/watch?v=wrCpiRqVczo

The issue appears to happen only with custom SPARK classes, and only with additional inventory slots. The standard SPARK class works fine in all respects.

Armor and primary weapon slots work fine, presumably secondary weapon slot works too. 

Other slots can be changed in the loadout screen, but as soon as you back out (e.g. to squad select screen), slots reset to their defaults. 

These happens at least with templated CHL slots from SPARK Arsenal.

**Comment by Iridar** [comment]
Done some investigating, and the claim turned out to be correct, though it had nothing to do with my mods.

The issue is caused by using the vanilla version of the [Custom SPARK Class Support](https://steamcommunity.com/sharedfiles/filedetails/?id=729610025) instead of  [Custom SPARK Class Support WOTC](https://steamcommunity.com/sharedfiles/filedetails/?id=1128879172) 

This is super curious, because the WOTC version of the mod is just the vanilla mod recompiled with WOTC SDK, which normally wouldn't make a difference to how the mod works.

Both versions do `UnitState.SetStatus(eStatus_Healing)` in `X2DLCInfo::OnPostMission()`. 

However, through some logging and script-tracing through a local Highlander I found out that the vanilla version of the mod sets `eStatus_Training` instead, which made the Squad Select screen think that the SPARK is in training, and unequipped all items from it, replacing them with the default loadout of infinite items.

I used UE Explorer to examine the compiled code of both mods, and turned out that in vanilla `UnitState.SetStatus(eStatus_Healing)` compiles to `UnitState.SetStatus(5)`, while in WOTC it is `UnitState.SetStatus(6)`. Evidently, because WOTC has more member in `eStatus` enumerable.

Mystery solved, case is closed. 

This is a very interesting thing to keep in mind when talking about using vanilla mods in WOTC, though.
